pkg/operator/hive/manageddns.go

```diff
@@ -37,38 +42,105 @@ func (r *ReconcileHiveConfig) teardownLegacyExternalDNS(hLog log.FieldLogger) er
 	return nil
 }
 
-func deployManagedDomainsConfigMap(h *resource.Helper, instance *hivev1.HiveConfig) error {
-	cm := &corev1.ConfigMap{}
-	cm.Kind = "ConfigMap"
-	cm.APIVersion = "v1"
-	cm.Name = managedDomainsConfigMapName
-	cm.Namespace = constants.HiveNamespace
+func (r *ReconcileHiveConfig) teardownLegacyMangedDomainsConfigMap(hLog log.FieldLogger) error {
+	key := client.ObjectKey{Namespace: constants.HiveNamespace, Name: "managed-domains"}
+	if err := resource.DeleteAnyExistingObject(r, key, &corev1.ConfigMap{}, hLog); err != nil {
+		hLog.WithError(err).Error("failed to delete legacy managed domains configmap")
+	}
+	return nil
+}
+
+// configureManagedDomains will create a new configmap holding the managed domains settings (if necessary), or simply
+// return the current configmap of the current deployment if the settings it contains match the desired settings.
+// It will also return the old configmap if a new one was created (so that it can be garbage collected).
```

Comment:
The function never returns the old configmap. It returns either the new one or the existing retained one.

Reply:
yep, that comment was for a previous version of behavior. will fix.